### PR TITLE
Vasilis/dml docs

### PR DESCRIPTION
**Pull request by vasilismsr**
Many changes to the dml user guide.

Also fixed bug #157 related to the model_final of KernelDMLCateEstimator.

Also added some docstrings to the statsmodelsinference classes and fixed docstring of the _RLearner class.

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -4,148 +4,170 @@
 # https://docs.microsoft.com/azure/devops/pipelines/languages/python
```
Comment (kbattocchi) [inline azure-pipelines.yml]: Why are these changes being made?

Comment (vasilismsr) [reply azure-pipelines.yml]: No idea! This was the result of merging with your commits. It had some conflicts there and I accepted the incoming change.

Comment (vasilismsr) [reply azure-pipelines.yml]: I figured these were the changes related to running linting only once.

Comment (kbattocchi) [reply azure-pipelines.yml]: I've pushed to this branch and reverted this back.

**Comment by kbattocchi** [comment]
Looks good.  I'll merge it in.

**Pull request merged**